utcp.c

```diff
@@ -1177,21 +1486,22 @@ bool utcp_is_active(struct utcp *utcp) {
 }
 
 struct utcp *utcp_init(utcp_accept_t accept, utcp_pre_accept_t pre_accept, utcp_send_t send, void *priv) {
-	struct utcp *utcp = calloc(1, sizeof *utcp);
-	if(!utcp)
-		return NULL;
-
 	if(!send) {
 		errno = EFAULT;
 		return NULL;
 	}
 
+	struct utcp *utcp = calloc(1, sizeof *utcp);
+	if(!utcp)
+		return NULL;
+
 	utcp->accept = accept;
 	utcp->pre_accept = pre_accept;
 	utcp->send = send;
 	utcp->priv = priv;
-	utcp->mtu = 1000;
-	utcp->timeout = 60;
+	utcp->mtu = DEFAULT_MTU;
+	utcp->timeout = DEFAULT_USER_TIMEOUT; // s
+	utcp->rto = START_RTO; // us
```

Comment:
On Thu, Oct 29, 2015 at 07:42:17AM -0700, vollkorn1982 wrote:

> > +	utcp->timeout = DEFAULT_USER_TIMEOUT; // s
> > +	utcp->rto = START_RTO; // us
> 
> What is meant by `s` and `us`?

Seconds and microseconds. I'll expand it.

-- 
Met vriendelijke groet / with kind regards,
     Guus Sliepen <guus@tinc-vpn.org>